packages/core/src/browser/statusbar/statusbar-entry-service.ts

```diff
@@ -0,0 +1,79 @@
+/*
+* Copyright (C) 2017 TypeFox and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+import { injectable, inject } from "inversify";
+import { StatusbarEntry } from './statusbar';
+import { CommandService } from '../../common';
+
+export interface StatusbarEntryAttributes {
+    className?: string;
+    title?: string;
+    onclick?: () => void;
+    onmouseover?: () => void;
+    onmouseout?: () => void;
+}
+
+@injectable()
+export class StatusbarEntryService {
+
+    constructor( @inject(CommandService) protected readonly commands: CommandService) { }
+
+    createAttributes(entry: StatusbarEntry): StatusbarEntryAttributes {
+        const attrs: StatusbarEntryAttributes = {
+            className: 'element'
+        };
+
+        if (entry.command) {
+            attrs.onclick = () => {
+                if (entry.command) {
+                    const args = entry.arguments || [];
+                    this.commands.executeCommand(entry.command, ...args);
+                }
+            };
+            attrs.onmouseover = () => {
+                attrs.className += ' hover';
+            };
+            attrs.onmouseout = () => {
+                attrs.className = 'element';
+            };
+        }
+
+        if (entry.tooltip) {
+            attrs.title = entry.tooltip;
+        }
+
+        return attrs;
+    }
+
+    mixupTextAndIcons(textArr: string[], iconClassArr: string[]): string[] {
```

Comment:
We discussed offline to extract this logic to a `LabelParser` component.
It should expose a single method 
```
   parse(text: string): LabelPart[]
```
where LabelPart is defined as follows:
```
type LabelPart = string | IconPart
interface IconPart {
    name: string
    animation?: string
}
```

Reply:
this can go into its own module, without any dom deps and therefore can be tested easily.